backend/src/main/java/ai/verta/modeldb/versioning/blob/container/EnvironmentContainer.java

```diff
@@ -220,12 +220,16 @@ private void validateEnvironmentVariableName(String name) throws ModelDBExceptio
 
   private String computeSHA(EnvironmentBlob blob) throws NoSuchAlgorithmException {
     StringBuilder sb = new StringBuilder();
-    sb.append("env:");
-    for (EnvironmentVariablesBlob environmentVariablesBlob : blob.getEnvironmentVariablesList()) {
-      sb.append(":name:")
-          .append(environmentVariablesBlob.getName())
-          .append("value:")
-          .append(environmentVariablesBlob.getValue());
+    List<
+            ai.verta.modeldb.versioning.autogenerated._public.modeldb.versioning.model
+                .EnvironmentVariablesBlob>
+        environmentVariables =
+            ai.verta.modeldb.versioning.autogenerated._public.modeldb.versioning.model
+                .EnvironmentBlob.fromProto(blob)
+                .getEnvironmentVariables();
+    if (environmentVariables != null) {
+      sb.append("env:");
+      sb.append(new HashSet<>(environmentVariables).hashCode());
```

Comment:
Are you sure Hash set will guarantee you the same order across restarts? We want the same set of envVars to map to same SHA.

Why not use environmentVariables itself? It is getting sorted in the auto gen code.

Reply:
oh sorry. Forgot about that. Fixed it. Also added a small change into template to use setters instead of set operator to force sort in fromProto.